language-plutus-core/src/Crypto.hs

```diff
@@ -1,15 +1,19 @@
 module Crypto (verifySignature) where
 
+import           Control.Applicative
 import           Crypto.ECC.Ed25519Donna
 import           Crypto.Error            (maybeCryptoError)
 import qualified Data.ByteString.Lazy    as BSL
-import           PlutusPrelude
 
--- This should return @error {bool}@ when it fails (and the spec should be updated
--- to reflect this); this code will be changed
-verifySignature :: BSL.ByteString -- ^ Public Key
-                -> BSL.ByteString -- ^ Message
-                -> BSL.ByteString -- ^ Signature
-                -> Bool
-verifySignature pubKey msg sig = fromMaybe False $
-    maybeCryptoError (verify <$> publicKey (BSL.toStrict pubKey) <*> pure (BSL.toStrict msg) <*> signature (BSL.toStrict sig))
+verifySignature
+    :: Alternative f
+    => BSL.ByteString  -- ^ Public Key
+    -> BSL.ByteString  -- ^ Message
+    -> BSL.ByteString  -- ^ Signature
+    -> f Bool
+verifySignature pubKey msg sig =
+    maybe empty pure . maybeCryptoError $
```

Comment:
I'd probably just pattern match.

Reply:
I think, `maybe empty pure` is readable and a pattern match would be heavy here. We could use one of my favorite operations: 

```haskell
single :: (Foldable f, Alternative g) => f a -> g a
single = foldr (const . pure) empty
```

which generalizes `eitherToList`, `listToMaybe`, `eitherToMaybe`, `maybeToList` and plenty of other operations.